### PR TITLE
Add ML_train workflow

### PR DESCRIPTION
**Pull request by tvoigtlaender**
Add parts of ML training of NMSSM analysis as ML_train workflow.
Add BaseWRoot env and add matplotlib to ML_train env.
Use law.SiblingFileCollection as default (please tell me if this breaks anything).
Multiple other quality of life changes.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -61,73 +61,91 @@ action() {
     fi
     
     #Determine which environment to use based on the luigi.cfg file
-    export ENV_NAME=$(grep '^ENV_NAME' lawluigi_configs/${ANA_NAME}_luigi.cfg | sed 's@ENV_NAME\s*=\s*\([^\s]*\)\s*@\1@g')
-    echo "Using ${ENV_NAME} environment."
+    export ENV_NAMES=($(grep '^ENV_NAME' lawluigi_configs/${ANA_NAME}_luigi.cfg | sed 's@ENV_NAME\s*=\s*\([^\s]*\)\s*@\1@g'))
```
Comment (harrypuuter) [inline setup.sh:64]: seems that is only works for bash and not for zsh

File: processor/framework.py [hunk 2]
```diff
@@ -43,12 +61,16 @@ def convert_env_to_dict(self, env):
                     pass
         return my_env
 
+    # Function to apply a source-script and get the resulting environment.
+    #   Anything apart from setting paths is likely not included in the resulting envs.
     def set_environment(self, sourcescript):
+        source_string = " ".join(sourcescript)
```
[the remark this answers is not in the document]
Comment (tvoigtlaender) [reply processor/framework.py:70]: I added a check for strings to both the set_environment and the run_command functions.

**Comment by tvoigtlaender** [comment]
The XRootD source script that was previously used seems to have some issues. I replaced it with the default KingMaker env source script when copying in tarballs of envs with no cvmfs version.

**Comment by harrypuuter** [comment]
ok nice, the `mdm1` problems still persist, but other than that, everything works with CROWN as well

**Pull request merged**